### PR TITLE
Use Vanilla application layout

### LINKED ISSUE #719: Unable to close sidebar in model details page
**Describe the bug**

From within the model details page, hover over the sidebar then click the user name in the bottom. It'll pop up, then down again. If you click it again it'll open but then you won't be able to close the sidebar again.

**To Reproduce**
1. Go to the model details page and hover over the sidebar
2. Click your username, it'll bounce.
3. Click it again to open it.
4. Click away from the sidebar, it'll stick around.

**Expected behavior**
It should close when not hovered.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Firefox
 - Version 81.0.2

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Use Vanilla application layout

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- [Add additional steps]

## Details

Vanilla ref: https://vanillaframework.io/docs/examples/layouts/application-structure
Zeplin ref: https://zpl.io/awRqDOj

## Screen recording

https://share.getcloudapp.com/JrugOBNL

Fixes #719

### CODE REVIEW

File: src/components/PrimaryNav/_primary-nav.scss [hunk 1]
```diff
@@ -108,13 +85,27 @@
     }
   }
 
-  .p-modal .p-list__link:hover {
-    text-decoration: underline;
+  .p-modal {
+    // Temporary override for the 'back to old gui' popup.
```
Comment (hatched) [inline src/components/PrimaryNav/_primary-nav.scss:89]: You can remove this and the link, the old GUI is no longer functional.

File: src/scss/functions/_z-index.scss [hunk 2]
```diff
@@ -16,6 +16,10 @@ $z-layers: (
   "iota": 8,
   "kappa": 9,
   "lambda": 10,
+  // Should be greater than anything set by third party (e.g. Vanilla)
+  "zelda": 99999,
```
Comment (hatched) [inline src/scss/functions/_z-index.scss:20]: botw?

**Review by hatched — APPROVED** [review]
Looks great, just if you could remove that 'back to old gui' link 👍

**Comment by hatched** [comment]
QA Notes:

- [ ] The cookie warning sticks to the bottom of the table, not fixed to the window.
- [ ] When resizing to the mobile view the sidebar pops out then disappears. (You can see it in your demo video as well).
- [ ] When in mobile view the 'add a controller' UI should be full width.
- [ ] The breakpoints in the Model Details view needs a bit of work, we should collapse the sidebar before the tables overflow the page.
- [ ] In mobile view the Model Details panels should be full width.
- [ ] In tablet view the Model Details panels should probably be full width? See image below.
Screen Shot 2020-10-23 at 4 20 37 PM

File: src/scss/functions/_z-index.scss (hunk 2 above)
Comment (barrymcgee) [reply src/scss/functions/_z-index.scss:20]: OoT

**Comment by spencerbygraves** [comment]
Thanks @barrymcgee

- As @hatched mentioned, I think the panel could be full width on the smallest screens, as on the [example below 375px](https://vanillaframework.io/docs/examples/layouts/application-structure)
- You can close a panel by clicking away but not the left nav. Would that need to updated in the layout?

**Comment by spencerbygraves** [comment]
@barrymcgee, just noticed that on small screen the top of the panel is cut off and you can't scroll up to view the name:

image

**Comment by barrymcgee** [comment]
> When resizing to the mobile view the sidebar pops out then disappears. (You can see it in your demo video as well).

This a side effect of the slide transition I've added to the side nav on small resolutions, the Vanilla pattern is a simple toggle appear/disappear. I think this is an acceptable trade off.

**Comment by ziheliu214** [comment]
@barrymcgee thanks Barry, I think it works perfectly.

a minor issue noticed (Maybe not really related to this issue):

When table headers in the panel truncated, the sort chevron is not displayed anymore.
image

**Pull request merged**